todo/src/containers/TodoList.js

```diff
@@ -0,0 +1,62 @@
+import React from "react";
+import { connect } from "react-redux";
+
+import { addNewTodo, toggleTodo } from "../actions/index";
+
+class TodoList extends React.Component {
+    state = {
+        newTodo: ""
+    };
+
+    handleChanges = e => {
+        this.setState({ newTodo: e.target.value });
+    };
+
+    addTodo = e => {
+        e.preventDefault();
+        this.props.addNewTodo(this.state.newTodo);
+    };
+
+    toggleTodo = (e, index) => {
+        e.preventDefault();
+        this.props.toggleTodo(index);
+    };
+
+    
+
+    render() {
+        return (
+            <>
+                <h1>ToDo List!!</h1>
+                <div className="list">
+                    {this.props.todos.map((todo, index) => (
+                        <h3
+                            className={todo.completed ? "completed" : ""}
+                            onClick={e => this.toggleTodo(e, index)}
+                            key={index}
+                        >
+                            {todo.value}
+                        </h3>
+                    ))}
+                </div>
+                <input 
+                    type="text"
+                    value={this.state.newTodo}
+                    placeholder="Todo"
+                    onChange={this.handleChanges}
+                />
+                <button onClick={this.addTodo}>Add Todo</button>
+            </>
+                
+        );
+    }
+}
+
+const mapStateToProps = state => ({
+    todos: state.todos,
+});
+
+export default connect(
+    mapStateToProps,
+    { addNewTodo, toggleTodo}
+)(TodoList);
```

Comment:
This is a cool approach. My only suggestion is to remember it should be in an unordered list per the instructions and I would have broken this down further to a functional component for the view. These are both small things, your code looks good!